baremetal/metal3machine_manager.go

```diff
@@ -558,6 +558,9 @@ func (m *MachineManager) Delete(ctx context.Context) error {
 			// We have no control over provisioning, so just wait until the
 			// host is powered off.
 			waiting = host.Status.PoweredOn
+		case bmov1alpha1.StatePreparing, bmov1alpha1.StateProvisioning, bmov1alpha1.StateProvisioned,
+			bmov1alpha1.StateDeprovisioning, bmov1alpha1.StatePoweringOffBeforeDelete,
+			bmov1alpha1.StateDeleting:
```

Comment:
Let's add a comment here at least that waiting = true for these, or add the literal waiting = true. Then L549 can be made `var waiting bool`. Let's also add default case, which defaults to true as well.